File: quickstart-python/services/queue_service.py
import json

from monday_code import PublishMessageParams, QueueApi

from models import APITypes
from services import with_monday_api, StorageService, MailService, LogsService


class QueueService:
    @staticmethod
    @with_monday_api(APITypes.QUEUE, "publish_message")
    async def publish_message(message: dict, api_instance: QueueApi = None) -> None:
        """
        This function will publish a message to our queue
        """
        publish_message_params = PublishMessageParams(
            message=json.dumps(message))
        return await api_instance.publish_message(publish_message_params)

    @staticmethod
    async def parse_queue_message(message: dict) -> None:
        """
        This function will parse the message from the queue to do some long-running process according to it's content
        """
        # Some long-running process, for example, sending an email
        # todo: Implement better parsing and handling of the message
        await LogsService.debug(f"Received message: {message}")
        message = json.loads(message.get('content'))
        if message.get('method') == 'send_mail':
            monday_access_token = message.get('user_token')
            storage_service = StorageService(monday_access_token)
            mail_address = await storage_service.get('mail_address')
            content = message.get('content')
            await MailService.send_mail(mail_address, content)

File: quickstart-node/src/services/monday-code/queue.service.ts
import { Queue } from '@mondaycom/apps-sdk';
import { MondayCodeQueueManager, QueueMessage } from '@my-types/monday-code-sdk';
import { LoggerService } from '@services/monday-code/logger.service';
import { QueueMethods } from '@shared/queue.consts';
import { StorageService } from '@services/monday-code/storage.service';
import { MailService } from '@services/mail.service';

export class QueueService implements MondayCodeQueueManager {
  // eslint-disable-next-line no-use-before-define
  private static instance: QueueService;
  private mondayCodeQueueManager;

  private constructor() {
    this.mondayCodeQueueManager = new Queue();
  }

  static getInstance(): QueueService {
    if (!this.instance) {
      this.instance = new QueueService();
    }

    return this.instance;
  }

  publishMessage(message: QueueMessage, options?: { topicName: string }): Promise<string> {
    return this.mondayCodeQueueManager.publishMessage(JSON.stringify(message), options);
  }

  validateMessageSecret(secret?: string): boolean {
    return this.mondayCodeQueueManager.validateMessageSecret(secret);
  }

  /**
   * This function will parse the message from the queue to do some long-running process according to its content.
   * This is only an example function, feel free to change it according to your needs
   */
  async parseQueueMessage(message: QueueMessage): Promise<void> {
    LoggerService.getInstance().debug(`Received message: ${JSON.stringify(message)}`);

    if (message.method === QueueMethods.SEND_EMAIL) {
      const mondayAccessToken = message.userToken;
      if (mondayAccessToken) {
        const mailAddress = (await new StorageService(mondayAccessToken).get<string>('mailAddress'))?.value;
        const content = message.content;
        if (mailAddress && content) {
          await MailService.sendMail(mailAddress, content);
        }
      }
    }
  }
}
